stackoverflow/src/main/java/ro/utcn/sd/cata/stackoverflow/controller/ConsoleController.java

```diff
@@ -0,0 +1,227 @@
+        print("User removed.");
+    }
+
+    private void handleLogin(){
+        print("Enter username:");
+        String username = scanner.next().trim();
+        print("Enter password:");
+        String password = scanner.next().trim();
+        Optional<User> user = userService.findUserByUsernamePassword(username,password);
+        if(user.equals(Optional.empty())){
+            print("User or Password incorrect. Please try again");
+        }
+        else{
+            loggedUser =  user.get();
+            loggedIn = true;
+            print("Login successful. Welcome "+loggedUser.getUsername()+"! \n");
+        }
+    }
+
+    private void handleAddQuestion(){
+        if(loggedIn){
+            String getNewlineRemainedFromPreviousCommand = scanner.nextLine().trim();
+            print("Enter title:");
+            String title = scanner.nextLine().trim();
+            print("Enter text:");
+            String text = scanner.nextLine().trim();
+            Question addedQuestion = questionService.addQuestion(title,text,loggedUser);
+            boolean doneAddingTags = false;
+            while(!doneAddingTags){
+                print("Enter tag or type done: ");
+                String tag = scanner.next().trim();
+                if(tag.equals("done")){
+                    doneAddingTags=true;}
+                else{
+                    tagService.addTag(tag,addedQuestion);}
+            }
+            print(("Question added: "+ addedQuestion.toString()));
+        }
+        else print("Log in in order to add questions.");
+    }
+
+    private void handleList(){
+        if(loggedIn){
+            for(Question question: questionService.findAllOrderByDate())
+                print(question.toString());
+        }
+        else print("Log in in order to see questions.");
+    }
+
+    private void handleFindTag(){
+        if(loggedIn){
+            print("Write the tag: ");
+            String tag = scanner.next().trim();
+            if(tagService.findByName(tag).isPresent()){
+                for(Question question: questionService.findByTag(tagService.findByName(tag).get())){
+                    print(question.toString());
+                }
+            }
+            else{
+                print("No questions found with tag "+tag);
+            }
+        }
+        else print("Log in in order to search questions.");
+    }
+
+    private void handleFindTitle(){
+        if(loggedIn){
+            print("Write title: ");
+            String title = scanner.next().trim();
+            List<Question> foundQuestions = questionService.findByTitle(title);
+            if(foundQuestions.isEmpty()){
+                print("No questions found with title containing "+title);
+            }
+            else
+            {
+                for(Question question: foundQuestions){
+                    print(question.toString());}
+            }
+        }
+        else print("Log in in order to search questions.");
```

Comment:
Generally it is advisable to always use curly brackets, even if you only have one statement in the if, else, for, etc. https://stackoverflow.com/a/8020255/7612556

Also, use the code formatter in your IDE please :)